src/tests/test_e2e_install.py

```diff
@@ -29,3 +37,86 @@ def test_networking(self, cluster, network_type):
     def test_olm_operator(self, cluster, olm_operators):
         cluster.prepare_for_installation()
         cluster.start_install_and_wait_for_installed()
+
+    @JunitTestSuite()
+    def test_upgrade_agent(self, cluster):
+        """
+        This test will prepare the cluster with the default agent image. Once it read to install it will restart the
+        service with an agent image that doesn't exist, so that all the hosts will move to `insufficient`. Then it will
```

Comment:
Actually the upgrade mechanism doesn't have the "older" or "newer" concept; it just upgrades from version _a_ to version _b_, where _b_ may be newer or older than _a_.

This test is checking the upgrade from _latest_ to _doesnotexist_, which fails, and then from _doesnotexist_ to _latest_, which works. 

It would be possible to check from an older version to a newer one, but that would require having access to those two versions of the agent images, i.e., the test would need to have two configuration parameters, something like `AGENT_DOCKER_IMAGE_OLD` and `AGENT_DOCKER_IMAGE`. As currently the test uses `:latest` instead of a numbered version I assumed that our jobs don't know anything about agent version numbers either.

Another alternative would be to change the build pipelines so that they publish the current `:latest` image and also a `:previoius` image (which doesn't really need to be the previous version, just a different tag). Then we could write a test to check the upgrade from `:previous` to `:latest`.

To summarize, with the current images that we publish and use in the tests I think that what this test does is reasonable, but I am open to implement one of the above alternatives. Let me know what you think.

Reply:
I also think that what the test does is sufficient with that regards.

Reply:
We can just take some existing image tag and use it for the test.
There is a difference in the upgrade flow if the new required image exists (the agent should successfully pull it and restart) vs the case image doesn't exist where the agents will move to insufficient but the actual agent upgrade flow isn't really tested.

Reply:
Changed the test so that it finds the previous image (the newest `vX.YZ` tag that has a different digest) and uses it initially. It then waits for nodes to be ready, restarts with a bad image, waits for nodes to be insufficient, and finally restart with the current image and waits for nodes to be ready.